spec/jobs/stream_equal_opportunities_report_publication_job_spec.rb

```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+
+RSpec.describe StreamEqualOpportunitiesReportPublicationJob do
+  subject(:job) { described_class.perform_later }
+  let(:vacancy) { create(:vacancy) }
+  let(:publish_report?) { true }
+  let(:event_data) { {} }
+
+  before do
+    vacancy.create_equal_opportunities_report
+    allow(Vacancy).to receive_message_chain(:expired_yesterday, :listed).and_return([vacancy])
+    allow(vacancy).to receive(:publish_equal_opportunities_report?).and_return(publish_report?)
```

Comment:
what's the harm in not stubbing these things?

Reply:
If these methods break I want the test that fails to be the most useful test: the unit tests for those methods. This StreamEqualOpportunitiesReportPublicationJob shouldn't know about/depend on the internals of Vacancy.

Reply:
Hmmm this feels a bit philosophical. Generally I feel wary of stubbing things too much, besides external services. 

I'd be interested to hear the rest of the team's thoughts on it!

Reply:
@dfe-teaching-vacancies How do you feel about this?

Reply:
correct me if i am wrong but i have viewed all tests apart from system specs as unit tests and to be true unit tests id say they should be stubbed. confused by connors comments a little `what's the harm in not stubbing these things?` -> `Generally I feel wary of stubbing things too much` as these arent services? i could be misunderstanding tho!

Reply:
that comment probably doesnt help much - making tests effective trumps being purist imo